### DISCUSSION TITLE
Terrain elevation - is it possible?

### DISCUSSION THREAD

**Discussion by issimissimo** [rnmapbox/maps#1431] [Q&A]
Hello, I'm totally new to Mapbox (I'm used to work with CesiumJS), and expecially to this version for React Native.
I'd really like to have satellite imagery with elevation, to be able to draw polylines on top of the 3D terrain and query for elevation data on some point, just like in this example for JS: "https://docs.mapbox.com/mapbox-gl-js/example/query-terrain-elevation/".
Is it possible to do that on mobile (Android / iOS) in general, and more specifically with this version for React Native?
P.S. I'm not asking HOW to do that, just IF it's possible, before to dig into the documentation, to spend a lot of time in trials and errors, and maybe to discover that's not possible at all, and so to be forced to come back to Cesium for React Native (Resium).
Many thanks!!!

**Comment by issimissimo** [comment]
I've read this article https://www.mapbox.com/blog/mapbox-mobile-sdk-v10-beta
about the 3d terrain, so I suppose it's quite a new feature, on mobile but on JS too (https://www.mapbox.com/blog/mapbox-gl-js-v2-3d-maps-camera-api-sky-api-launch)
At this point I think the answer to my question is "NO".
I really like Mapbok for its smoothness, and its out of the box place labels feature, so I rephrase my question:
is there maybe some hope to see this new version to be implemented in react-native-mapbox-gl in an acceptable time?
Again, many thanks!

**Comment by marcibk** [comment]
Isn´t it just possible to use the iOS / android version and create a native component with the 3d terrain? Thinking about trying it out.

**Comment by issimissimo** [comment]
Really a good idea! Unfortunately I'm not able at all to create native components...

**Comment by nzjs** [comment]
There is a discussion around this here:
https://github.com/react-native-mapbox-gl/maps/discussions/1202
